Kros.Utils/src/Kros.Utils.MsAccess/Data/BulkActions/MsAccess/MsAccessBulkInsert.cs

```diff
@@ -214,19 +212,30 @@ public void Insert(IBulkActionDataReader reader)
             }
         }
 
-        /// <summary>
-        /// Vloží všetky dáta zo zdroja <paramref name="reader"/>.
-        /// </summary>
-        /// <param name="reader">Zdroj dát.</param>
-        public void Insert(IDataReader reader)
+        /// <inheritdoc/>
+        public async Task InsertAsync(IBulkActionDataReader reader)
+        {
+            using (var bulkInsertReader = new BulkActionDataReader(reader))
+            {
+                await InsertAsync(bulkInsertReader);
+            }
+        }
+
+        /// <inheritdoc/>
+        public void Insert(IDataReader reader) => InsertCoreAsync(reader, useAsync: false).GetAwaiter().GetResult();
+
+        /// <inheritdoc/>
+        public Task InsertAsync(IDataReader reader) => InsertCoreAsync(reader, useAsync: true);
+
+        private async Task InsertCoreAsync(IDataReader reader, bool useAsync)
```

Comment:
Here the methods `CreateDataFile` and `InitBulkInsert` can be async too - writing the files.

Reply:
Yes. I also think about it, but:

The main reason we do this is to support `KORM` in `ASP.NET` api, where it is important to be able to call `async` methods. With `ASP.NET`, We use `MsSQL` instead of `MsAccess`. When implementing `async` in `MsAccess`, I chose the easier way. Because I'm not sure how it is with context switching, because a large number of rows would call `WriteAsync`.